### PR TITLE
Add support for package and module in Pylint

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Add support for package and module in Pylint.

In python, a project can look like:
```
src
└── package
    ├── __init__.py
    ├── sub_package1
    │   ├── __init__.py
    │   ├── module1.py
    │   └── module2.py
    └── sub_package2
        ├── __init__.py
        ├── module3.py
        └── module4.py
```

So if a file named `__init__.py` exists in a folder, python will consider that folder as a package (or sub-package, but it's the same as a package, so no big difference). And any file that ends in `.py` is called a module.

Since warnings-ng supports package and module, I decided to also add support for them in the pylint parser.

### CODE REVIEW

File: src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java [hunk 1]
```diff
@@ -219,6 +219,24 @@ void shouldParseReportWithoutSymbol() {
...
+                    .hasPackageName("python_package");
+            softly.assertThat(iterator.next())
+                    .hasLineStart(1)
+                    .hasLineEnd(1)
+                    .hasMessage("Unused import os (unused-import)")
+                    .hasFileName("trunk/src/module_name_no_package.py")
+                    .hasCategory("W0611")
+                    .hasSeverity(Severity.WARNING_NORMAL)
+                    .hasModuleName("module_name_no_package")
+                    .hasPackageName(null);
```
Comment (JeanChristopheMorinPerso) [inline src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java]: This specific test fails and I have absolutely no idea why. If anyone have ideas, I'm all ear :ear: 
I'm not a Java developer, so I might be missing something obvious...

**Comment by JeanChristopheMorinPerso** [comment]
PS: I still have to check with other tools than Pylint if what I did is consistent.

File: src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java (hunk 1 above)
Comment (uhafner) [reply src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java]: `null` is not possible. Seems that the parser returns the module name.

File: src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java [hunk 2]
```diff
@@ -41,6 +41,14 @@ protected boolean isLineInteresting(final String line) {
...
         builder.setCategory(StringUtils.firstNonBlank(matcher.group("symbol"), category, UNKNOWN_CAT));
 
+        final String moduleName = matcher.group("module");
+        if (moduleName != null) {
+            if (moduleName.contains(".")) {
+                builder.setPackageName(moduleName.substring(0, moduleName.lastIndexOf(".")));
+            }
+            builder.setModuleName(moduleName);
+        }
+
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java]: Here you need:
```
else {
   builder.setPackageName("-").setModuleName("-");
}
```

Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java]: The builder is not cleared between warnings.

File: src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java (hunk 1 above)
Comment (uhafner) [reply src/test/java/edu/hm/hafner/analysis/parser/PylintParserTest.java]: `null` is not allowed, use `"-"`

File: src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java]: Aaa, that confirms what I was seeing, effectively. I had the impression something wasn't cleared, but didn't knew what it was. I'll adjust this, thanks :)

Comment (JeanChristopheMorinPerso) [reply src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java]: Fixed.

**Pull request merged**